File: src/main/java/cc/sukazyo/cono/morny/bot/event/OnUniMeowTrigger.java
package cc.sukazyo.cono.morny.bot.event;

import cc.sukazyo.cono.morny.bot.api.EventListener;
import cc.sukazyo.cono.morny.bot.command.ISimpleCommand;
import cc.sukazyo.cono.morny.util.tgapi.InputCommand;
import com.pengrad.telegrambot.model.Update;

import javax.annotation.Nonnull;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicBoolean;

public class OnUniMeowTrigger extends EventListener {
	
	private static final Map<String, ISimpleCommand> triggers = new HashMap<>();
	
	public static void register (ISimpleCommand... list) {
		for (ISimpleCommand cmd : list)
			triggers.put(cmd.getName(), cmd);
	}
	
	@Override
	public boolean onMessage (@Nonnull Update event) {
		if (event.message().text() == null) return false;
		AtomicBoolean ok = new AtomicBoolean(false);
		triggers.forEach((name, command) -> {
			name = "/" + name;
			if (name.equals(event.message().text())) {
				command.execute(new InputCommand(name), event);
				ok.set(true);
			}
		});
		return ok.get();
	}
	
}

File: src/main/scala/cc/sukazyo/cono/morny/bot/event/OnUniMeowTrigger.scala
package cc.sukazyo.cono.morny.bot.event

import cc.sukazyo.cono.morny.bot.api.{EventEnv, EventListener}
import cc.sukazyo.cono.morny.bot.command.MornyCommandManager
import cc.sukazyo.cono.morny.util.tgapi.InputCommand
import cc.sukazyo.cono.morny.Log.logger

class OnUniMeowTrigger (using commands: MornyCommandManager) extends EventListener {
	
	override def onMessage (using event: EventEnv): Unit = {
		import event.*
		
		givenCxt >> { (input: InputCommand) =>
			logger trace s"got input command {$input} from event-context"
			
			for ((name, command_instance) <- commands.commands_uni) {
				logger trace s"checking uni-meow $name"
				if (name == input.command)
					logger trace "checked"
					command_instance.execute(using input, event.update)
					event.setEventOk
			}
			
		} || { logger trace "not command (for uni-meow)" }
		
	}
	
}
